src/components/Advertising/handlers/viewThroughHandler.js

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2025 Adobe. All rights reserved.
+Licensed under the Apache License, Version 2.0.
+http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+import collectAllIdentities from "../identities/collectAllIdentities.js";
+import {
+  LOG_ERROR_RESOLVING_ID,
+  LOG_AD_CONVERSION_FAILED,
+  AD_CONVERSION_VIEW_EVENT_TYPE,
+} from "../constants/index.js";
+import {
+  isAnyIdUnused,
+  appendAdvertisingIdQueryToEvent,
+  markIdsAsConverted,
+} from "../utils/helpers.js";
+
+export default async function handleViewThrough({
+  eventManager,
+  cookieManager,
+  logger,
+  componentConfig,
+  adConversionHandler,
+}) {
+  const resolvedIds = {};
+  const usedIdTracker = {};
+  const conversionTasks = [];
+
+  const triggerConversion = async () => {
+    if (!isAnyIdUnused(resolvedIds, usedIdTracker)) return null;
+
+    const idTypesToUse = Object.keys(resolvedIds);
+
+    try {
+      const event = appendAdvertisingIdQueryToEvent(
+        resolvedIds,
+        eventManager.createEvent(),
+        cookieManager,
+        componentConfig,
+        true,
+      );
+      const xdm = {
+        eventType: AD_CONVERSION_VIEW_EVENT_TYPE,
+      };
+      event.setUserXdm(xdm);
+
+      const result = await adConversionHandler.trackAdConversion({ event });
+
+      markIdsAsConverted(idTypesToUse, usedIdTracker, cookieManager, logger);
+      return result;
+    } catch (error) {
+      logger.error(LOG_AD_CONVERSION_FAILED, error);
+      return null;
+    }
+  };
+
+  const identityPromisesMap = collectAllIdentities(
+    logger,
+    componentConfig,
+    cookieManager,
+  );
+
+  if (Object.keys(identityPromisesMap).length === 0) {
+    return [];
+  }
+
+  const identityPromises = Object.entries(identityPromisesMap).map(
+    ([idType, idPromise]) =>
+      idPromise
+        .then((idValue) => {
+          if (idValue) {
+            resolvedIds[idType] = idValue;
+            const conversionPromise = triggerConversion();
```

Comment:
You could just do "return triggerConversion();" here. If you return a promise from inside a then, the resulting promise will wait for all to be resolved. Then you can get rid of the conversionTasks array.

Reply:
```suggestion
            return triggerConversions();
```

`triggerConversions` will always return a truthy value (a Promise is truthy), and the following line `if` is testing the promise, not what the promise contains.

If you return the promise from this `.then` function, then you can also make lines 88-89 just

```js
  return Promise.allSettled(identityPromises)
```

and delete the entire `conversionTasks` array.